src/v/cluster/topic_table.cc

```diff
@@ -24,6 +25,14 @@
 
 namespace cluster {
 
+topic_table::topic_table(ss::sharded<partition_manager>& pm)
+  : _probe(*this)
+  , _pm(pm) {
+    register_delta_notification([this](const std::vector<delta>& deltas) {
+        _pm.local().notify_topic_table_deltas(deltas);
+    });
+};
```

Comment:
> We route the deltas via partition_manager to avoid entangled dependency issues between partition and topic_table lifecycles.

i don't fully understand the reasoning behind routing through the partition manager. is it because the caller has a reference to the partition manager and cannot include the topics_table header file?

Reply:
This approach may be a bit stale now, I need to take a look at it again. Some back story,

When I began with the PR, I wanted to plumb the topic deltas into a partition (`tm_stm`) so that I can perform some action on delete notifications. The problem there is the init order `partition_manager-> partition -> tm_stm` where as topic table inits via `partition_manager -> controller-> topic_table` (controller needs an active partition_manager instance), so we may need some additional coordination to access `topic_table` to make sure it is inited before downstream callers of partition_manager can use it. This seemed complex. Rather an easier way is to pass a partition manager hook into topic_table that is called on init.

That said, as the patch evolved, I ended up redoing the notifications inside `tx_gateway_frontend` (not `tm_stm`) which is inited late in the bootstrap, so it should have access to `controller` instance, so I can probably avoid this commit.

(There is a dependent transactions change that Vadim is working on that might affect the direction of this PR, so I'm holding off for now to see where that goes, will update this accordingly, but feel free to poke holes in ^^^ meanwhile).

Reply:
can we simply use `manage/unmange` notifications from `partition_manager` ? the same way we are doing in `kafka::group_manager` ?

Reply:
```
    /*
     * register for notification of partitions within the specific topic
     * being removed from manager. this will invoke the callback for existing
     * partitions synchronously so the caller must be prepared for that.
     *
     * the callback must not block.
     *
     * we don't currently have any mechanism for un-managing partitions, so that
     * interface is non-existent.
     */
    notification_id_type register_unmanage_notification(
```